File: army_challenge/services/castle.py
from ..types import Civilization, Army
from ..services import ArmiesRegistry, training_camp

class Castle:

    @staticmethod
    def create_army(civilizationType:Civilization, army: Army) -> str:
        try:
            ArmiesRegistry.registryArmy(army)
            TrainingCamp.createUnits(civilizationType,army)
            army.set_army_strength(army)
            return "Ejército creado exitosamente"
        except Exception as error:
            return f"Error creating army: {error}"

File: src/services/castle.ts
import type { Civilization, Army } from '../types/index'
import { ArmiesRegistry } from '../services/armiesRegistry';
import { TrainingCamp } from '../services/trainingCamp';

//Clase helper que sólo agrupa lógica.
export class Castle {

    static createArmy(civilizationType:Civilization, army: Army):string{
        try {
            ArmiesRegistry.registryArmy(army)
            TrainingCamp.createUnits(civilizationType,army)
            army.setArmyStrength();
            return "Army created successfully"
        } catch (error) {
            return `Error creating army: ${error}`
        }
    }

}
